src/EurekaClient.js

```diff
@@ -303,19 +303,21 @@ export default class Eureka {
    */
   transformRegistry(registry) {
     if (!registry) {
-      throw new Error('Unable to transform empty registry');
-    }
-    this.cache = { app: {}, vip: {} };
```

Comment:
Actually there's no issue with this, because JS will execute this synchronously. Nothing can interrupt the event loop.